### PR TITLE
add FAN_MIN_PWM for slow fan stopping issue

### LINKED ISSUE #1980: main fan stops at low PWM speeds
The main fan stops running at low PWM speeds (K8200 at about 35 to 40 of 255), even if kickstarted with FAN_KICKSTART_TIME.

I would say this is a bug (even if hardware dependent) because the user - and host software or slicer - would expect the fan to run if set to speeds higher than 0.

So the fan should be driven to run at least at minimal possible speed when fan speed is set to 1 or other "to slow" value.

Can be easiely reproduced: Start fan manually, lower speed to small values greater 0, fan will go slower and slower until stop. You can hear PWM sounds from fan indicating that it is still powered.

### PR DESCRIPTION
**Pull request by CONSULitAS**
closes #1980

The main fan stops running at low PWM speeds (K8200 at about 35 to 40 of 255), even if kickstarted with FAN_KICKSTART_TIME.

If fan speed is lower than FAN_MIN_PWM (and not 0) it is set (dynamically) to FAN_MIN_PWM to run at least on minimal possible speed.

### CODE REVIEW

**Comment by AnHardt** [comment]
Makes no sense to me. In ether case you have to find a value fitting your system. A year later, when the fan is worn, you need to flash an other value. I think, it's better the user knows about a lower limit for the PWM.

**Comment by CONSULitAS** [comment]
@AnHardt see answer on #1980 , too.

But if you know this lower limit, you can set it to FAN_MIN_PWM for your convenience and you are sure, your fan won't stop.

File: Marlin/Configuration.h [hunk 1]
```diff
@@ -43,7 +48,7 @@ Here are some standard links for getting your machine calibrated:
 #define STRING_VERSION "1.0.3 dev"
 #define STRING_URL "reprap.org"
 #define STRING_VERSION_CONFIG_H __DATE__ " " __TIME__ // build date and time
-#define STRING_CONFIG_H_AUTHOR "(none, default config)" // Who made the changes.
+#define STRING_CONFIG_H_AUTHOR "(K8200, CONSULitAS)" // Who made the changes.
```
Comment (thinkyhead) [inline Marlin/Configuration.h:50]: Oops, you don't want to overwrite the main `Configuration.h`…

File: Marlin/Configuration_adv.h [hunk 2]
```diff
@@ -191,7 +196,7 @@
 //homing hits the endstop, then retracts by this distance, before it tries to slowly bump again:
 #define X_HOME_BUMP_MM 5
 #define Y_HOME_BUMP_MM 5
-#define Z_HOME_BUMP_MM 2
+#define Z_HOME_BUMP_MM 3
```
Comment (thinkyhead) [inline Marlin/Configuration_adv.h:199]: And you probably want to revert this line too.

File: Marlin/planner.cpp [hunk 3]
```diff
@@ -440,12 +440,17 @@ void check_axes_activity() {
         } else {
           fan_kick_end = 0;
         }
-    #endif//FAN_KICKSTART_TIME
+    #endif //FAN_KICKSTART_TIME
+    #ifdef FAN_MIN_PWM
+      #define CALC_FAN_SPEED ((tail_fan_speed == 0) ? 0 : ( (tail_fan_speed < FAN_MIN_PWM) ? FAN_MIN_PWM : tail_fan_speed ) )
```
Comment (thinkyhead) [inline Marlin/planner.cpp:444]: I kind of like the idea of having "1" correspond to the FAN_MIN_PWM, have "255" always correspond to 255, and then scale all other values proportionally. So, the calculation would then be:

```
tail_fan_speed ? FAN_MIN_PWM + (tail_fan_speed * (255 - FAN_MIN_PWM)) / 255 : 0;

// Examples:
//   80 +   1 * (255 - 80) / 255 =  80
//   80 + 128 * (255 - 80) / 255 = 167
//   80 + 255 * (255 - 80) / 255 = 255
```

**Comment by thinkyhead** [comment]
After you patch this up, try creating a new branch from it, squashing all the commits except one, and re-submitting the PR. It will be much cleaner in the commit log.

**Comment by CONSULitAS** [comment]
@thinkyhead you are right, to late last night :-1: 

will implement your idiea for linear behavior

will make a new one later today and close this

**Comment by CONSULitAS** [comment]
will be remade!

**Pull request closed without merging**